### ISSUE TITLE
PyTorch 2.4.0 & last pytorch_dlprim:  UserWarning: The operator 'aten::log_sigmoid_forward' is not currently supported

### ISSUE THREAD

**Issue by sukamenev** [artyom-beilis/pytorch_dlprim#82]
/home/.../ve_root/lib/python3.11/site-packages/torch/nn/functional.py:3244: UserWarning: The operator 'aten::log_sigmoid_forward' is not currently supported on the ocl backend. Please open an issue at for requesting support https://github.com/artyom-beilis/pytorch_dlprim/issues (Triggered internally at /home/.../pytorch_dlprim/src/tensor_ops.cpp:334.)
  return torch.binary_cross_entropy_with_logits(input, target, weight, pos_weight, reduction_enum)
  
 It's very strange: in PyTorch 1.13.1 & previsious pytorch_dlprim I don't get this warning.

**Comment by artyom-beilis** [comment]
I think I implemented logits few hours ago... can you retest? Also now there is a binary releases so you can easily install it with pytorch 2.4

**Comment by sukamenev** [comment]
Same error. May be you forgot remove warning?

/home/.../ve_root/lib/python3.11/site-packages/torch/nn/functional.py:3244: UserWarning: The operator 'aten::log_sigmoid_forward' is not currently supported on the ocl backend. Please open an issue at for requesting support https://github.com/artyom-beilis/pytorch_dlprim/issues (Triggered internally at /home/.../pytorch_dlprim/src/tensor_ops.cpp:334.)
  return torch.binary_cross_entropy_with_logits(input, target, weight, pos_weight, reduction_enum)
  
  I see this warning only one time. After the warning, the program works fine.
  In any case, the accuracy of the model increases.

**Comment by artyom-beilis** [comment]
> `aten::log_sigmoid_forward` 
and
> `binary_cross_entropy_with_logits`
I see. Maybe in 1.13 it was implemented in terms of other ops and 2.4 it got its own operator.

I suggest take a look on implementation of `binary_cross_entropy` - I assume it would be easy to create similar operator.

Currently I'm debugging YOLO issues and it would take me few days to get into it. You may try to implement one on your own. Anyway since it is only final loss you shouldn't suffer from major performance issues - it just fallbacks to CPU.

**Comment by artyom-beilis** [comment]
If you don't manage to implement it. Ping me in a week or so

**Comment by sukamenev** [comment]
> I think I implemented logits few hours ago

А где они были реализованы в dlprimitives или в pytorch_dlprim?

**Comment by artyom-beilis** [comment]
> А где они были реализованы в dlprimitives или в pytorch_dlprim?

Look there: https://github.com/artyom-beilis/pytorch_dlprim/blob/master/src/loss_ops.cpp#L90
